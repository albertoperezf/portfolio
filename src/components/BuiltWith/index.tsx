import Image from "next/image";
import styles from "@/styles/Home.module.css";

export default function BuiltWith(): JSX.Element {
  return (
    <div>
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        target="_blank"
        rel="noopener noreferrer"
      >
        Build with
        <Image
          className={styles.logo}
          src="/next.svg"
          alt="Next.js Logo"
          width={70}
          height={24}
          priority
        />
        By{" "}
        <Image
          src="/vercel.svg"
          alt="Vercel Logo"
          className={styles.vercelLogo}
          width={70}
          height={24}
          priority
        />
      </a>
    </div>
  );
}
