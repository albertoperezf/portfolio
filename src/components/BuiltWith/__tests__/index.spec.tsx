import { render, screen } from "@testing-library/react";
import BuiltWith from "..";

describe("BuiltWith component", () => {
  it('should render a link with target="_blank" and rel="noopener noreferrer"', () => {
    render(<BuiltWith />);
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("target", "_blank");
    expect(link).toHaveAttribute("rel", "noopener noreferrer");
  });

  it("should render the Next.js logo with width=70 and height=24", () => {
    render(<BuiltWith />);
    const nextLogo = screen.getByAltText("Next.js Logo");
    expect(nextLogo).toBeInTheDocument();
    expect(nextLogo).toHaveAttribute("src", "/next.svg");
    expect(nextLogo).toHaveAttribute("width", "70");
    expect(nextLogo).toHaveAttribute("height", "24");
  });

  it("should render the Vercel logo with width=70 and height=24", () => {
    render(<BuiltWith />);
    const vercelLogo = screen.getByAltText("Vercel Logo");
    expect(vercelLogo).toBeInTheDocument();
    expect(vercelLogo).toHaveAttribute("src", "/vercel.svg");
    expect(vercelLogo).toHaveAttribute("width", "70");
    expect(vercelLogo).toHaveAttribute("height", "24");
  });

  it("should match snapshot", () => {
    const { container } = render(<BuiltWith />);
    expect(container).toMatchSnapshot();
  });
});
