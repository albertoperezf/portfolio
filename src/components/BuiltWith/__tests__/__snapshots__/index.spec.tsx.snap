// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuiltWith component should match snapshot 1`] = `
<div>
  <div>
    <a
      href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      Build with
      <img
        alt="Next.js Logo"
        class="logo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="24"
        src="/next.svg"
        style="color: transparent;"
        width="70"
      />
      By
       
      <img
        alt="Vercel Logo"
        class="vercelLogo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="24"
        src="/vercel.svg"
        style="color: transparent;"
        width="70"
      />
    </a>
  </div>
</div>
`;
